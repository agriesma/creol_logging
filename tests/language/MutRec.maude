load creol-interpreter .
mod PROGRAM is
  protecting CREOL-SIMULATOR .
  op state : -> State [ctor] .
  eq state = {
    < "MutRec" : Class | Inh: noInh, Param: noVid, Att: noSubst,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "_" |-> null, Code: skip ;
              return ( emp ) >,
            < "doit" : Method | Param: "other", Att: "other" |-> null,
              "ccglab0" |-> null, "_" |-> null, Code: call( "ccglab0" ;
              "other" ; "doit" ; "this" ) ; get( "ccglab0" ; noVid ) ;
              free( "ccglab0" ) ; return ( emp ) >,
      Ocnt: 0 >
    
    < "Test" : Class | Inh: noInh, Param: noVid, Att: "m1" |-> null, "m2" |->
      null,
      Mtds: < "init" : Method | Param: noVid, Att: "_" |-> null,
              Code: new( "m1" ; "MutRec" ; emp ) ;
              new( "m2" ; "MutRec" ; emp ) ; return ( emp ) >,
            < "run" : Method | Param: noVid, Att: "ccglab0" |-> null, "_" |->
              null, Code: call( "ccglab0" ; "m1" ; "doit" ; "m2" ) ;
              free( "ccglab0" ) ; return ( emp ) >,
            < "doit" : Method | Param: "other", Att: "other" |-> null, "_"
              |-> null, Code: skip ; return ( emp ) >,
      Ocnt: 0 > } .
endm

