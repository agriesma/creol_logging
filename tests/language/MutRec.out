==========================================
reduce in PROGRAM : state .
result State: {
< "MutRec" : Class | Inh: noInh, Param: noVid, Att: noSubst, Mtds:
  < "doit" : Method | Param: "other", Att: "_" |-> null, "ccglab0" |-> null,
    "other" |-> null, Code: call("ccglab0" ; "other" ; "doit" ; "this") ; get(
    "ccglab0" ; noVid) ; free("ccglab0") ; return(emp) >, < "init" : Method |
    Param: noVid, Att: "_" |-> null, Code: skip ; return(emp) >, < "run" :
    Method | Param: noVid, Att: "_" |-> null, Code: skip ; return(emp) >,
  Ocnt: 0 >
 
< "Test" : Class | Inh: noInh, Param: noVid, Att: "m1" |-> null, "m2" |-> null,
    Mtds:
  < "doit" : Method | Param: "other", Att: "_" |-> null, "other" |-> null,
    Code: skip ; return(emp) >, < "init" : Method | Param: noVid, Att: "_" |->
    null, Code: new("m1" ; "MutRec" ; emp) ; new("m2" ; "MutRec" ; emp) ;
    return(emp) >, < "run" : Method | Param: noVid, Att: "_" |-> null,
    "ccglab0" |-> null, Code: call("ccglab0" ; "m1" ; "doit" ; "m2") ; free(
    "ccglab0") ; return(emp) >,
  Ocnt: 0 >
}
Bye.
